Standards/CCSMB-12.md

```diff
@@ -0,0 +1,64 @@
+| MIME        | `model/3dja`              |
+| Extensions  | `.3dja`                   |
+
+## Technical details
+
+The file MUST be composed of a single JSON object with this basic ruleset:
+- The writer SHOULD specify global settings for the printed 3D blocks in the global namespace of the object, if a global setting isn't set each block model MUST have it set individually.
+- The writer MAY define a dictionary of presets (which are tint/texture combinations), if this isn't done every block MUST use their own texture and tint.
+- The writer MUST define an array of block models, following the specifications of the [3dj](https://docs.sc3.io/features/sc-peripherals.html#_3dj-format) format with the addition of a coordinates array (if the user foregoes presets each block model object should be compatible with a 3dj file if the global settings are inserted, the coordinates field WILL be disregarded).
+- Each block model MAY override a global setting.
+The following represents the file format in a more visual aspect
+```jsonc
+{
+  "label": "...",         // (RECOMMENDED) The name of the build, if specified end result of a block in the build SHALL be "BuildName: blocks[i].label", if not end result SHALL be blocks[i].label
+  "tooltip": "...",       // (OPTIONAL) The description of the build, shared between all blocks unless overridden
+  "size": [1,1,1],        // (RECOMMENDED) Size (x,y,z) in number of blocks required for the build, not needed for printing but is an indicator for an automated builder
+  "isButton": false,      // Global setting for all blocks, if not assigned every block must have it's own setting
+  "collideWhenOn": true,  // Global setting for all blocks, if not assigned every block must have it's own setting
+  "collideWhenOff": true, // Global setting for all blocks, if not assigned every block must have it's own setting
+  "lightLevel": 0,        // Global setting for all blocks, if not assigned every block must have it's own setting
+  "redstoneLevel": 0,     // Global setting for all blocks, if not assigned every block must have it's own setting
+  "presets": {            // (OPTIONAL) Dictionary of presets, used to define the texture and tint of a shape for compresion
+    "presetname": {
+      "texture": "",      // (REQUIRED) Texture of the preset
+      "tint": "FFFFFF"    // (REQUIRED) Tint of the preset
+    },
+  },
+  "blocks": [
+    {
+      "label": "...",             // (OPTIONAL) The name of the block, if empty SHALL use "(blocks[i].x,blocks[i].y,blocks[i].z)" as blocks[i].label
+      "tooltip": "...",           // (OPTIONAL) The description of the block, overrides the tooltip of the build
+      "coordinates": [[0, 0, 0]], // Order: x,y,z. 0,0,0 represents the origin of the build, used for automated builders and desambiguation, each set of coordinates SHOULD be globally unique (if not, automated building programs SHALL fail), the block MUST be printed for the amount of coordinates in this array
```

Comment:
array of an array? clarify why this is please
also, use SHOULD instead of SHALL

Reply:
SHALL is also used in RFC.
array of array allows you to place the same block at multiple parts of the build, you can figure out the amount of blocks to print by printing the amount of coordinates

Reply:
fair